### PR TITLE
IA-2397 Use terra-app-setup-chart and persist PVC ids

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2397
https://broadworkbench.atlassian.net/browse/IA-2533

Automation test failed becuz https://github.com/broadinstitute/firecloud-develop/pull/2426 needs to be merged

* use set up chart
* Persist pvc ids

Validated persisting pvc part working properly in fiab
```
qi@cloudshell:~ (qi-billing-2)$ kubectl get pvc -n ya8b8r-gxy-ns ya8b8r-gxy-rls-galaxy-pvc -o=jsonpath='{.metadata.uid}'
7d5ab7f8-5b75-422d-b4f5-0ee1b93a404e
qi@cloudshell:~ (qi-billing-2)$ kubectl get pvc -n ya8b8r-gxy-ns ya8b8r-gxy-rls-cvmfs-alien-cache-pvc -o=jsonpath='{.metadata.uid}'
939ec32a-6e8e-4a63-b6b5-a6802412bb9f
qi@cloudshell:~ (qi-billing-2)$
```
and this is what's in leo db

```
| id | nodepoolId | appType | appName | status  | samResourceId                        | creator                           | createdDate                | destroyedDate              | dateAccessed               | namespaceId | diskId | customEnvironmentVariables | googleServiceAccount                                               | kubernetesServiceAccount | chart                      | release        | galaxyPvcId                          | cvmfsPvcId                           |
+----+------------+---------+---------+---------+--------------------------------------+-----------------------------------+----------------------------+----------------------------+----------------------------+-------------+--------+----------------------------+--------------------------------------------------------------------+--------------------------+----------------------------+----------------+--------------------------------------+--------------------------------------+
|  1 |          4 | GALAXY  | pvc-0   | RUNNING | 5ca2d63e-59fb-4979-b7b7-10b905a75f5d | hermione.owner@test.firecloud.org | 2021-02-18 17:24:54.945000 | 1970-01-01 00:00:01.000000 | 2021-02-18 17:24:54.945000 |           1 |      1 | NULL                       | b153pet-110530393451290928813@qi-billing-2.iam.gserviceaccount.com | gxy-ksa                  | galaxy/galaxykubeman-0.7.2 | ya8b8r-gxy-rls | 7d5ab7f8-5b75-422d-b4f5-0ee1b93a404e | 939ec32a-6e8e-4a63-b6b5-a6802412bb9f |
```

TODO:
* Move pvc ids to PD table
* Add `lastUsedBy` in PD table
* 
---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by jdcanas** [comment]
This all looks like it will play well with the custom app code im working on, with a lot of conflicts though :(

File: Dockerfile [hunk 1]
```diff
@@ -33,7 +33,15 @@ RUN curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master
 # Add the repos containing nginx and galaxy charts
 RUN helm repo add center https://repo.chartcenter.io && \
     helm repo add galaxy https://raw.githubusercontent.com/cloudve/helm-charts/anvil/ && \
+    helm repo add terra-app-setup-charts https://storage.googleapis.com/terra-app-setup-chart && \
```
Comment (rtitle) [inline Dockerfile:36]: Did you push this manually?

I had created a `terra-app-charts` bucket, wondering if we could use that bucket for both the terra-app and setup charts.

File: Dockerfile [hunk 2]
```diff
@@ -33,7 +33,15 @@ RUN curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master
...
     helm repo update
 
+# .Files helm helper can't access files outside a chart. Hence in order to populate cert file properly, we're
+# pulling `terra-app-setup` locally and add cert files to the chart.
+# Leonardo will install the chart from local version.
+RUN pushd /leonardo && \
```
Comment (rtitle) [inline Dockerfile:42]: Could also `cp` the certs here instead of mounting them separately in the docker-compose file

Comment (rtitle) [reply Dockerfile:42]: Er actually maybe not.. the cp would run before the mount I guess

File: Dockerfile (hunk 1 above)
Comment (Qi77Qi) [reply Dockerfile:36]: yea..I manually pushed to this chart....I wasn't sure if it's supposed to be in the same bucket and was afraid to break the existing chart so I created a separate one

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/ConfigImplicits.scala [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import pureconfig.ConfigReader
+import cats.syntax.all._
+import org.broadinstitute.dsp.{ChartName, ChartVersion}
+import pureconfig.error.ExceptionThrown
+
+import java.nio.file.{Path, Paths}
+
+object ConfigImplicits {
```
Comment (rtitle) [inline core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/ConfigImplicits.scala:10]: Just added because `Config.scala` is getting large? Should we move all the implicits here?

File: Dockerfile (hunk 2 above)
Comment (Qi77Qi) [reply Dockerfile:42]: yea, `cp` is what I did the first time...it doesn't work becuz the files don't exist yet

File: http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210217_add_app_pvc_id.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="leonardo" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet logicalFilePath="leonardo" author="qi" id="add_app_pvc_id">
+        <addColumn tableName="APP">
+            <column name="galaxyPvcId" type="varchar(254)">
```
Comment (rtitle) [inline http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210217_add_app_pvc_id.xml:6]: This is probably fine for now, but we should think about data model for `CUSTOM` app type. Maybe we want some different tables containing Galaxy-specific fields.

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/ConfigImplicits.scala (hunk 3 above)
Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/ConfigImplicits.scala:10]: I meant to talk about this during walkthrough but we didn't have time...But basically I'm introducing pureconfig in this PR....For new configs, I think we should just use `pureconfig`, which will be less manual. These are implicits for primitive types that can be shared by different projects (when we move multi project structure)

File: Dockerfile (hunk 1 above)
Comment (rtitle) [reply Dockerfile:36]: I think the same bucket can have multiple charts

File: http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210217_add_app_pvc_id.xml (hunk 4 above)
Comment (Qi77Qi) [reply http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210217_add_app_pvc_id.xml:6]: I see...we can probably wait until Justin's PR merged and revisit

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppComponent.scala [hunk 5]
```diff
@@ -181,6 +188,16 @@ object appQuery extends TableQuery(new AppTable(_)) {
       .map(_.status)
       .update(status)
 
+  def updatePvcIds(id: AppId, galaxyPvcId: PvcId, cvmfsPvcId: PvcId)(implicit ec: ExecutionContext): DBIO[Unit] =
+    for {
+      _ <- getByIdQuery(id)
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppComponent.scala]: I think you can update in 1 query instead of 2:
```
getByIdQuery(id).map(x => (x.a, x.b)).update((Some(a), Some(b)))
```

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 6]
```diff
@@ -305,32 +296,42 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       gkeClusterId = dbCluster.getGkeClusterId
       googleProject = params.googleProject
 
-      _ <- logger.info(ctx.loggingCtx)(
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:308]: Can we keep this log statement?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 7]
```diff
@@ -305,32 +296,42 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
...
-        s"Beginning app creation for app ${app.appName.value} in cluster ${gkeClusterId.toString}"
-      )
-
       // Create namespace and secrets
       _ <- logger.info(ctx.loggingCtx)(
         s"Creating namespace ${namespaceName.value} and secrets for app ${app.appName.value} in cluster ${gkeClusterId.toString}"
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: Maybe remove `and secrets` from this log

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 8]
```diff
@@ -305,32 +296,42 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
...
+          s"Cluster not found in Google: ${gkeClusterId} | trace id: ${ctx.traceId}"
+        )
       )
-      _ <- kubeService.createServiceAccount(gkeClusterId, ksa, KubernetesNamespace(namespaceName))
 
+      nfsDisk <- F.fromOption(
+        dbApp.app.appResources.disk,
+        AppCreationException(s"NFS disk not found in DB for app ${app.appName.value} | trace id: ${ctx.traceId}")
+      )
+
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:330]: Maybe add a log "installing set-up chart"

(There are a lot of steps here and I find logging each step useful)

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 9]
```diff
@@ -387,6 +374,13 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       _ <- logger.info(ctx.loggingCtx)(
         s"Finished app creation for app ${app.appName.value} in cluster ${gkeClusterId.toString}"
       )
+      pvcs <- kubeService.getPersistentVolumeClaims(gkeClusterId, KubernetesNamespace(app.appResources.namespace.name))
+      galaxyPvc = pvcs.find(pvc => pvc.getMetadata.getName == s"${app.release.asString}-galaxy-pvc")
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: Are these suffixes defined by the Galaxy chart?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 10]
```diff
@@ -387,6 +374,13 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
...
+      cvmfsPvc = pvcs.find(pvc => pvc.getMetadata.getName == s"${app.release.asString}-cvmfs-alien-cache-pvc")
+      _ <- (galaxyPvc, cvmfsPvc).tupled.traverse {
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: Wondering if we should throw an error if we can't retrieve the PVCs

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 9 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: yea, according to Alex, these are hardcoded

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 10 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: haha...I can totally do that...was being lazy and punted it lol

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 8 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:330]: there will be the following log already...do you still want the additional log? I found it a bit too noisy when each step prints out something
```
2021/02/18 17:29:55 creating 3 resource(s)
2021/02/18 17:29:55

Finished installing release:  ya8b8r-gxy-rls-setup-rls
{"@timestamp":"2021-02-18T17:29:55.075Z","@version":"1","message":"The command 'helm installChart' succeeded","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","thread_name":"ioapp-compute-4","severity":"INFO","level_value":20000}
```

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 6 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:308]: this log was right before logging `Creating namespace` xxx, there's nothing interesting happening between the 2 log lines..so I figured we don't need both of them. I can add `Begin app creation` text to the second line

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 11]
```diff
@@ -87,11 +79,13 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       nodepools = dbCluster.nodepools
         .filter(n => params.nodepoolsToCreate.contains(n.id))
         .map(buildLegacyGoogleNodepool)
+      _ = println(s"222 ${dbCluster.id} ${dbCluster.nodepools.map(_.id)} ${params.clusterId}")
```
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: reminder to remove before merging :)

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 6 above)
Comment (kyuksel) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:308]: FWIW, when looking through logs having app creation and namespace creation lines separate would make it easier for me to spot things.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 12]
```diff
@@ -387,6 +378,17 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
...
+      galaxyPvc = pvcs.find(pvc => pvc.getMetadata.getName == s"${app.release.asString}-galaxy-pvc")
+      cvmfsPvc = pvcs.find(pvc => pvc.getMetadata.getName == s"${app.release.asString}-cvmfs-alien-cache-pvc")
+      _ <- (galaxyPvc, cvmfsPvc).tupled
+        .fold(F.raiseError[Unit](new LeoException("Fail to retrieve pvc ids", traceId = Some(ctx.traceId)))) {
+          case (gp, cp) =>
+            appQuery
+              .updatePvcIds(params.appId, PvcId(gp.getMetadata.getUid), PvcId(cp.getMetadata.getUid))
+              .transaction
+              .void
+        }
```
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: Not sure if this logic will work. I think we need to distinguish between two possible cases and take different actions accordingly, i.e., 

1. When creating a Galaxy app **and** a **new** disk, do as above.
2. When creating a Galaxy app **reattaching** an **existing** disk, get the PVC IDs from `PERSISTENT_DISK` table (since we must have executed 1. at some point when first creating the disk)

[link to the related section of the design doc](https://broadworkbench.atlassian.net/wiki/spaces/IA/pages/1184464905/2021-01-05+Galaxy+Persistence#Explanation)

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: we don't have any code for re-attaching disk yet(it'll be in the next PR)...this is PR's goal is really just to retrieve the disk IDs and persist them (prepping for the actual re-attaching implementation)

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: actually, I might just update this to reflect re-attaching in the same PR...

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: I end up making this particular change since it doesn't inflate this PR that much...but don't worry too much about re-attaching logic yet in this PR since it'll be addressed in a follow up PR...

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 8 above)
Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:330]: I think I'd still like an additional `log.info`… the helm client logs tend to be in a different format I think and may be hard to parse (e.g. with `TraceId`).

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala [hunk 13]
```diff
@@ -89,6 +155,23 @@ object persistentDiskQuery extends TableQuery(new PersistentDiskTable(_)) {
...
+    findByIdQuery(id)
+      .map(x => (x.galaxyPvcId, x.cvmfsPvcId, x.chart, x.release))
+      .update(
+        (Some(galaxyDiskRestore.galaxyPvcId),
+         Some(galaxyDiskRestore.cvmfsPvcId),
+         Some(galaxyDiskRestore.chart),
+         Some(galaxyDiskRestore.release))
+      )
+
+  def isUsedByGalaxy(id: DiskId)(implicit ec: ExecutionContext): DBIO[Option[Boolean]] =
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala:164]: Wondering if this should return `DBIO[Boolean]`.. is `Option` important?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala [hunk 14]
```diff
@@ -168,7 +251,8 @@ object persistentDiskQuery extends TableQuery(new PersistentDiskTable(_)) {
       disk.size,
       disk.diskType,
       disk.blockSize,
-      disk.formattedBy
+      disk.formattedBy,
+      None //TODO: update this one PersistentDisk model is updated
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala:253]: Are you planning to do this? Or should we leave this out of the `PersistentDisk` model?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 6 above)
Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:308]: Am ok with combining the logs below.. mainly just wanted a "begin" signal in the logs.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala [hunk 15]
```diff
@@ -387,6 +381,33 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       _ <- logger.info(ctx.loggingCtx)(
         s"Finished app creation for app ${app.appName.value} in cluster ${gkeClusterId.toString}"
       )
+
+      // TODO: this logic will be changed in the next PR
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:385]: How will this change? 

Right now it looks like:
- if the disk already has a PvcId saved (e.g. it was previously attached to a Galaxy) --> do nothing
- if the disk does NOT have a PvcId saved, fetch it from k8s and persist it

**Comment by rtitle** [comment]
I think this looks pretty good so far.. cool that the app tests pass. (RStudio failed but maybe that is transient.)

A bit more needs to be done for persistence - do you prefer to keep adding to this PR, or do you want to try and get this merged as is and open subsequent PRs?

**Comment by Qi77Qi** [comment]
@rtitle if this is mostly good with you, less the restore part (that part will be changed anyway, but I didn't want to pile too many things in this PR), I'd like to have restore logic in a separate PR just so it's easier to review

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala (hunk 15 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:385]: it's mostly the criteria for determining whether we need to persist PVCIds could change. Since there's also `createDisk` flag, that could be used as well....I'm trying to not duplicating this info (also not duplicating `chart`, `release` in multiple places) somehow..I don't know exactly what it'll look like yet, hence just left a comment

**Comment by rtitle** [comment]
Sure works for me, as long as app creation works in this PR (the set-up chart stuff looks good).

BTW for persistence, I tagged you in https://github.com/galaxyproject/galaxykubeman-helm/pull/22 because it seemed relevant - we might need to get those `galaxykubeman` PRs merged and tagged to integrate with Leo..

**Review by rtitle — APPROVED** [review]
👍  once tests pass

**Comment by Qi77Qi** [comment]
oh yea...forgot about that chart PR....oh well

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala (hunk 13 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala:164]: this function will actually be removed in the follow up PR

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala (hunk 14 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala:253]: I'm not sure yet..but will revisit in the follow up PR

**Comment by Qi77Qi** [comment]
PR should be ready now....Tested in fiab that deleteApp works properly now. The previous bug was because setup chart was deleted before galaxy chart, which relies on a deletion job to run, which I'm guessing depends on ksa to exist. We don't currently catch this in automation tests, I could potentially add a test for this in the follow up PR, but also not sure if this will increase test instability.

When I was trying deleteApp in dev, I ran into the following error from `deleteNamespace` call.
```
{"response":"{\n  \"kind\": \"Status\",\n  \"apiVersion\": \"v1\",\n  \"metadata\": {\n    \n  },\n  \"status\": \"Failure\",\n  \"message\": \"Operation cannot be fulfilled on namespaces \\\"ehawk4-gxy-ns\\\": The system is ensuring all content is removed from this namespace.  Upon completion, this namespace will automatically be purged by the system.\",\n  \"reason\": \"Conflict\",\n  \"details\": {\n    \"name\": \"ehawk4-gxy-ns\",\n    \"kind\": \"namespaces\"\n  },\n  \"code\": 409\n}","result":"Failed"}"
```
vs, with this PR, we get 
```
{"@timestamp":"2021-02-24T01:53:52.406Z","@version":"1","message":"{\"response\":\"Some(())\",\"result\":\"Succeeded\"}","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","thread_name":"ioapp-compute-4","severity":"INFO","level_value":20000,"traceId":"b28422af-dd80-4f2c-8a80-0c2ce3e3c0aa","googleCall":"io.kubernetes.client.openapi.apis.CoreV1Api.deleteNamespace(ncenrf-gxy-ns, true, null, null, null, null, null)","duration":"39 milliseconds","result":"Succeeded"}
```


I think this might be due to previously we weren't deleting ksa, secrets explicitly before calling `deleteNamespace`, which I think this PR should make things better
